custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+            return None
+        else:
+            subtitle.content = response.content
+    
+    @staticmethod
+    def _is_string_japanese(string):
+        for char in string:
+            if ('\u3040' <= char <= '\u309F' or  # Hiragana
+                '\u30A0' <= char <= '\u30FF' or  # Katakana
+                '\u4E00' <= char <= '\u9FFF' or  # Kanji (CJK Unified Ideographs)
+                '\uFF66' <= char <= '\uFF9D'):   # Half-width Katakana
+                return True
+        return False
+    
+    @staticmethod
+    def _is_archive(archive: bytes):
+        archive_stream = io.BytesIO(archive)
+        
+        if rarfile.is_rarfile(archive_stream):
+            logger.debug("Identified rar archive")
+            return rarfile.RarFile(archive_stream)
+        elif zipfile.is_zipfile(archive_stream):
+            logger.debug("Identified zip archive")
+            return zipfile.ZipFile(archive_stream)
+        else:
+            logger.debug("Doesn't seem like an archive")
+            return None
+        
+    @staticmethod
+    def _subtitle_sorting_key(file):
+        name = file["name"].lower()
+        is_srt = name.endswith('.srt')
+        
+        # Usually netflix > webrip > amazon, with the rest having the lowest priority
+        sub_sources = ["netflix", "webrip", "amazon"]
+        priority = len(sub_sources)
+        for index, source in enumerate(sub_sources):
+            if source in name:
+                priority = index
+                break
+        
+        return (not is_srt, priority, file["name"])
+    
+    @cache
+    def _get_jimaku_response(self, url_path):
+        url = f"{self.api_url}/{url_path}"
+        
+        retry_count = 0
+        while retry_count < self.api_ratelimit_backoff_limit:
+            retry_count += 1
+            response = self.session.get(url, timeout=10)
+            
+            if response.status_code == 429:
+                api_reset_time = float(response.headers.get("x-ratelimit-reset-after", 5))
+                reset_time = self.api_ratelimit_max_delay if api_reset_time > self.api_ratelimit_max_delay else api_reset_time
+                
+                logger.warning(f"Jimaku ratelimit hit, waiting for '{reset_time}' seconds ({retry_count}/{self.api_ratelimit_backoff_limit} tries)")
+                time.sleep(reset_time)
+            else:
+                response.raise_for_status()
+            
+            data = response.json()
+            logger.debug(f"Length of response on {url}: {len(data)}")
+            if len(data) == 0:
+                logger.error(f"Jimaku returned no items for our our query: {url_path}")
+                return None
+            else:
+                return data
+            
+        # Max retries exceeded
+        raise TooManyRequests(f"Jimaku ratelimit max backoff limit of {self.api_ratelimit_backoff_limit} reached, aborting.")
```

Comment:
This is going to throttle for 1 hour, is this intentional? Check the get_providers.py `provider_throttle_map`

Reply:
There is also APIThrottled that seems more appropriated for what you want here but im not sure.

Reply:
Thanks for the heads up! Switched over to `APIThrottled` as that exception does seem more appropriate.